### DISCUSSION TITLE
yml based config

### DISCUSSION THREAD

**Discussion by afdesk** [smartforce-io/atc#9] [Ideas]
yml based config (per repository) - uses build in github workflow
## Config options
* type (maven, gradle, npm, custom regexp )
* behavior (version changed after actual modifications/version changed before actual modification)
* tag name template

### maven
it means `atc` should look for `pox.xml`, and then used `version` field.
### gradle
need to add
### npm
need to add

## Behaviors
* `after` - version changed after actual modifications
* `before` - version changed before actual modification

## tag name template
I suggest to use Go Template Format:
```yaml
template: v{{.version}}
```

## Sample
```yaml
type: maven
behavior: after
template: v{{.version}}
```
@AndreyLevchenko can you take a look at this discussion?

**Comment by AndreyLevchenko** [comment]
`behavior` is too general. We will have to come up with better name

**Comment by AndreyLevchenko** [comment]
and here is config I used for testing last time:
https://github.com/AndreyLevchenko/test-repo/blob/master/.atc.yaml
